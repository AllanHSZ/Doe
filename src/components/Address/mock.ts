import { AddressProps } from 'services/user'

export default [
  {
    id: 1,
    number: '1',
    region: 'LESTE',
    state: 'State 1',
    street: 'Street 1',
    neighborhood: 'Neighborhood 1',
    city: 'City 1'
  },
  {
    id: 2,
    number: '2',
    region: 'LESTE',
    state: 'State 2',
    street: 'Street 2',
    neighborhood: 'Neighborhood 2',
    city: 'City 2'
  },
  {
    id: 3,
    number: '3',
    region: 'LESTE',
    state: 'State 3',
    street: 'Street 3',
    neighborhood: 'Neighborhood 3',
    city: 'City 3'
  },
  {
    id: 4,
    number: '4',
    region: 'LESTE',
    state: 'State 4',
    street: 'Street 4',
    neighborhood: 'Neighborhood 4',
    city: 'City 4'
  },
  {
    id: 5,
    number: '5',
    region: 'LESTE',
    state: 'State 5',
    street: 'Street 5',
    neighborhood: 'Neighborhood 5',
    city: 'City 5'
  }
] as AddressProps[]
