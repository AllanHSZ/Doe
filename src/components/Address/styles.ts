import styled, { css, DefaultTheme } from 'styled-components'
import { AddressComponentProps } from '.'

type WrapperProps = Pick<AddressComponentProps, 'appearance'>

const wrapperModifiers = {
  full: (theme: DefaultTheme) => css`
    flex: 1;
    min-height: 26rem;

    ${ItemsWrapper} {
      flex: 1;
    }

    ${NotFound} {
      margin: auto;
    }

    & > button {
      margin-top: ${theme.spacings.small};
    }
  `,
  compact: (theme: DefaultTheme) => css`
    ${ItemsWrapper} {
      flex: 1;
    }

    ${NotFound} {
      background: #fff;
      padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
      border-radius: ${theme.border.radius.medium};
      box-shadow: ${theme.boxShadow.small};
    }

    & > button {
      margin-top: ${theme.spacings.xsmall};
    }
  `
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, appearance }) => css`
    display: flex;
    flex-direction: column;
    align-items: flex-end;

    ${wrapperModifiers[appearance!](theme)}
  `}
`

export const ItemsWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.xsmall};
    width: 100%;
  `}
`

export const NotFound = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`

export const Heading = styled.h5`
  ${({ theme }) => css`
    grid-area: heading;
    width: 100%;
    margin: ${theme.spacings.xxsmall} 0;
  `}
`

export const Form = styled.form`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.xsmall};
  `}
`

export const FormBody = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-wrap: wrap;
    gap: ${theme.spacings.xsmall};

    & > div {
      flex: 1;
      min-width: 40%;

      &:nth-child(1) {
        flex: 2 0 10rem;
      }

      &:nth-child(2) {
        flex: 1 0 6rem;
      }

      &:nth-child(3) {
        flex: 3 0 60%;
      }

      &:nth-child(4) {
        flex: 2 0 10rem;
      }

      &:nth-child(5) {
        flex: 1 0 6rem;
      }
    }
  `}
`

export const Footer = styled.div`
  ${({ theme }) => css`
    flex: 1;
    display: flex;
    min-width: 80%;
    justify-content: flex-end;
    gap: ${theme.spacings.xxsmall};
  `}
`
