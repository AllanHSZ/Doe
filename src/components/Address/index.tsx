import Button from 'components/Button'
import { Trash } from 'react-feather'
import * as S from './styles'

const Address = () => (
  <S.Wrapper>
    <S.ItemsWrapper>
      <S.AddressItem>
        <p>Av. XXX, 000 - Bairro Xx</p>
        <Trash size={18} />
      </S.AddressItem>
      <S.AddressItem>
        <p>Av. XXX, 000 - Bairro Xx</p>
        <Trash size={18} />
      </S.AddressItem>
    </S.ItemsWrapper>
    <Button appearance="outline">Adicionar</Button>
  </S.Wrapper>
)

export default Address
