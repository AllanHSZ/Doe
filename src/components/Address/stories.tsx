import { Story, Meta } from '@storybook/react/types-6-0'

import Address, { AddressComponentProps } from '.'
import mock from './mock'

export default {
  title: 'Address',
  component: Address
} as Meta

export const Default: Story<AddressComponentProps> = (args) => (
  <Address {...args} />
)

Default.args = {
  items: mock
}
