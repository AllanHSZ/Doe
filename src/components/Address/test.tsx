import { renderWithTheme } from 'utils/tests/helpers'

import Address from '.'

describe('<Address />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Address />)

    expect(true).toBeTruthy()

    expect(container.firstChild).toMatchSnapshot()
  })
})
