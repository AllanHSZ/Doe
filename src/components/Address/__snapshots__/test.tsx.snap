// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Address /> should render the heading 1`] = `
.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Lato,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  cursor: pointer;
  border-radius: 0.6rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #04D361;
  border: 2px solid currentColor;
  background: transparent;
  height: 4rem;
  font-size: 1.4rem;
  font-weight: 700;
  padding: 0.8rem 3.2rem;
}

.c3:hover {
  color: #04AB4F;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.8rem;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 1.6rem;
  gap: 0.8rem;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-size: 1.4rem;
  padding: 0.8rem 1.6rem;
  box-shadow: 0px 0px 4px rgba(0,0,0,0.2);
  background: #fff;
  gap: 2.4rem;
}

.c2 svg {
  color: #E33D3D;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <p>
        Av. XXX, 000 - Bairro Xx
      </p>
      <svg
        fill="none"
        height="18"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="3 6 5 6 21 6"
        />
        <path
          d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      <p>
        Av. XXX, 000 - Bairro Xx
      </p>
      <svg
        fill="none"
        height="18"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="3 6 5 6 21 6"
        />
        <path
          d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
        />
      </svg>
    </div>
  </div>
  <button
    class="c3"
  >
    <span>
      Adicionar
    </span>
  </button>
</div>
`
